# HG changeset patch
# User Rik <rik@octave.org>
# Date 1544731059 28800
#      Thu Dec 13 11:57:39 2018 -0800
# Node ID 1a12d4369beb7244e4bfa641493ebb8c12cf6edf
# Parent  07e60035b09bcbe1d14e9ed436125966394ed66a
Fix "Value" property for uicontrol togglebuttons (bug #55211).
Also fix "string" property for uicontrol pushbuttons.

* ToggleButtonControl.cc (update): If property is not handled by ToggleButton
update() function then forward event to ButtonControl::update.

* PushButtonControl.cc (update): Remove handling of "string" property.
If property is not handled by PushButton update() function then forward event
to ButtonControl::update.

diff --git a/libgui/graphics/PushButtonControl.cc b/libgui/graphics/PushButtonControl.cc
--- a/libgui/graphics/PushButtonControl.cc
+++ b/libgui/graphics/PushButtonControl.cc
@@ -69,29 +69,25 @@ namespace QtHandles
   void
   PushButtonControl::update (int pId)
   {
     uicontrol::properties& up = properties<uicontrol> ();
     QPushButton *btn = qWidget<QPushButton> ();
 
     switch (pId)
       {
-      case uicontrol::properties::ID_STRING:
-        btn->setText (Utils::fromStdString (up.get_string_string ()));
-        break;
-
       case uicontrol::properties::ID_CDATA:
         {
           octave_value cdat = up.get_cdata ();
           QImage img = Utils::makeImageFromCData (cdat,
                                                   cdat.rows (),
                                                   cdat.columns ());
           btn->setIcon (QIcon (QPixmap::fromImage (img)));
         }
         break;
 
       default:
-        BaseControl::update (pId);
+        ButtonControl::update (pId);
         break;
       }
   }
 
 }
diff --git a/libgui/graphics/ToggleButtonControl.cc b/libgui/graphics/ToggleButtonControl.cc
--- a/libgui/graphics/ToggleButtonControl.cc
+++ b/libgui/graphics/ToggleButtonControl.cc
@@ -86,14 +86,14 @@ namespace QtHandles
           QImage img = Utils::makeImageFromCData (cdat,
                                                   cdat.rows (),
                                                   cdat.columns ());
           btn->setIcon (QIcon (QPixmap::fromImage (img)));
         }
         break;
 
       default:
-        BaseControl::update (pId);
+        ButtonControl::update (pId);
         break;
       }
   }
 
 };
